bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -115,6 +115,20 @@ case class Proposal(meta:Meta,
 
   def resetObservationMeta:Proposal = copy(observations = observations.map(_.copy(meta = None)))
 
+  // Calculate the program time as a ratio of total time.
+  private def timeSum(extract: Observation => Option[TimeAmount]): TimeAmount =
+    TimeAmount.sum(observations.map(extract).flatten)
+
+  def programTimeHours: TimeAmount = timeSum(_.progTime)
+  def partnerTimeHours: TimeAmount = timeSum(_.partTime)
```

Comment:
Is there something mutable here that makes these be `def`s and not `val`s or do we just not want to store them?  If they were `val`s you wouldn't need temporary variables in the `programTimeRatio` method, which could be a `val` too come to think of it.